File: android/src/main/kotlin/com/example/video_trimmer/handlers/TrimVideoHandler.kt
package com.example.video_trimmer.handlers

import android.content.Context
import androidx.media3.common.util.UnstableApi
import com.example.video_trimmer.BaseMethodHandler
import com.example.video_trimmer.VideoManager
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import kotlinx.coroutines.CoroutineScope    
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext

@UnstableApi
class TrimVideoHandler(private val context: Context) : BaseMethodHandler {
    private val scope = CoroutineScope(Dispatchers.Main)

    override fun handle(call: MethodCall, result: MethodChannel.Result) {
        val startTimeMs = call.argument<Number>("startTimeMs")?.toLong() // Long for milliseconds
        val endTimeMs = call.argument<Number>("endTimeMs")?.toLong() // Long for milliseconds

        if (startTimeMs == null || endTimeMs == null) {
            result.error("INVALID_ARGUMENTS", "Missing startTimeMs/endTimeMs parameters", null)
            return
        }

        scope.launch {
            try {
                val path = VideoManager.getInstance(context).trimVideo(
                    startTimeMs = startTimeMs,
                    endTimeMs = endTimeMs
                )
                withContext(Dispatchers.Main) {
                    result.success(path)
                }
            } catch (e: Exception) {
                withContext(Dispatchers.Main) {
                    result.error("TRIM_ERROR", e.message, null)
                }
            }
        }
    }
}

File: ios/Classes/handlers/TrimVideoHandler.swift
import Flutter

class TrimVideoHandler: BaseMethodHandler {
    func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        guard let args = call.arguments as? [String: Any],
              let startTime = args["startTimeMs"] as? Int,
              let endTime = args["endTimeMs"] as? Int else {
            result(FlutterError(code: "INVALID_ARGUMENTS",
                              message: "Missing or invalid startTime/endTime parameters",
                              details: nil))
            return
        }
        
        let outputPath = args["outputPath"] as? String
        
        VideoManager.shared.trimVideo(
            startTimeMs: Int64(startTime),
            endTimeMs: Int64(endTime),
            outputPath: outputPath
        ) { trimResult in
            switch trimResult {
            case .success(let path):
                result(path)
            case .failure(let error):
                result(FlutterError(code: "TRIM_ERROR",
                                  message: error.localizedDescription,
                                  details: nil))
            }
        }
    }
}
